stablehlo/transforms/StablehloRefineShapes.cpp

```diff
@@ -316,6 +318,114 @@ LogicalResult evalElementwise(PatternRewriter& rewriter, OpType op,
   return success();
 }
 
+template <class AttrElementT, class TargetAttrElementT, class CalculationT,
+          typename OpType>
+LogicalResult evalConvertHelper(PatternRewriter& rewriter, OpType op,
```

Comment:
I'm curious if these are needed for shape refinement or are you using this pass for general op-folding? Asking since we have `StablehloAggressiveSimplification.cpp` which is where I think we should start moving folders to soon. Doesn't have to be a part of this PR, but AFAIK shape computations are usually integers (operating on dim sizes), but perhaps there's more to it

Reply:
Yes, this thought also came into my mind at first, while i was exploring available passes for model simplification. It wasn't obvious to me that `--stablehlo-refine-shapes` option also performs partial evaluation/specialization of some operators, until i looked into code. While i haven't examined the extent to which shape refinement relies on Eval* specialization, but would guess that it can be extracted into standalone pass with, maybe, `--stablehlo-apply-partial-evaluation/specialization` option.

Reply:
I'm still not sure of the best way to go about this sort of pattern without impacting shape refinement.

Two options I see:
1. If it's only a few pattern's we'll be duplicating float support for, perhaps we can just duplicate and rename the other to `EvalConvertOpIntPattern` which is used by shape refinement only
2. If there's a lot, we should make `evalElementwise` more verbose with FP/Int support and then experiment to see what works best --  having separate int/float patterns, having shape passes take an int flag on construction that says "int_only", etc

Reply:
i'd like to start with separate convert function using existing `constFoldCastOp` function allowing for early return with `castStatus`. Once there is a need for additional rewrite pattern to support float computations, then we can enchance `evalElementwise` with float support.